projects/kafka-cron/producer/internal/producer/producer.go

```diff
@@ -0,0 +1,91 @@
+package producer
+
+import (
+	"context"
+	"fmt"
+	"kafka-cron/pkg/models"
+	"time"
+
+	"encoding/json"
+
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+	"github.com/google/uuid"
+)
+
+func Producer(brokers string) (*kafka.Producer, error) {
+	p, err := kafka.NewProducer(&kafka.ConfigMap{
+		"bootstrap.servers": brokers,
+	},
+	)
+	if err != nil {
+		return nil, err
+	}
+	return p, nil
+}
+
+func CreateTopic(p *kafka.Producer, topic string) error {
+	a, err := kafka.NewAdminClientFromProducer(p)
+	if err != nil {
+		return fmt.Errorf("failed to create new admin client from producer: %s", err)
+	}
+	// Contexts are used to abort or limit the amount of time
+	// the Admin call blocks waiting for a result.
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	// Create topics on cluster.
+	// Set Admin options to wait up to 60s for the operation to finish on the remote cluster
+	maxDur, err := time.ParseDuration("60s")
+	if err != nil {
+		return fmt.Errorf("parseDuration(60s): %s", err)
+	}
+	results, err := a.CreateTopics(
+		ctx,
+		// Multiple topics can be created simultaneously
+		// by providing more TopicSpecification structs here.
+		[]kafka.TopicSpecification{
+			{
+				Topic:             topic,
+				NumPartitions:     2,
+				ReplicationFactor: 1,
+			},
+		},
+		// Admin options
+		kafka.SetAdminOperationTimeout(maxDur))
+	if err != nil {
+		return fmt.Errorf("admin Client request error: %v", err)
+	}
+	for _, result := range results {
+		if result.Error.Code() != kafka.ErrNoError && result.Error.Code() != kafka.ErrTopicAlreadyExists {
+			return fmt.Errorf("failed to create topic: %v", result.Error)
+		}
+		fmt.Printf("%v\n", result)
+	}
+	a.Close()
+	return nil
+}
+
+func ProduceMessage(p *kafka.Producer, topic string, job models.CronJob) error {
+	deliveryChan := make(chan kafka.Event)
+	message, err := json.Marshal(job)
+	if err != nil {
+		return fmt.Errorf("failed to marshal message: %v", err)
```

Comment:
```suggestion
		return fmt.Errorf("failed to marshal message: %w", err)
```

See https://stackoverflow.com/questions/61283248/format-errors-in-go-s-v-or-w